### PR TITLE
Adds Eof V1.0 (Eips: 3540, 3670, 4750, 4200, 5450)

### PR DESCRIPTION
**Pull request by Demuirgos**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.

## Changes:
- Implements Eip3540: 
    * Adds Eof format to bytecode and eof validation
- Implements Eip3670:
   * Adds Eof Body/Instructions validation
- Implements Eip4200:
   * Adds static jumps (RJUMP. RJUMPI, RJUMPV)
   * Extends Eof body validation to include new opcodes
- Implements Eip4750:
   * Adds Functions sections (CALLF, RETF)
   * Extends Eof body validation to include new opcodes
   * Extends Eof body validation to validate codesections
- Implements Eip5450:
   * Extends Eof body validation to validate stack requirements of opcodes/functions
   * Extends Eof body validation to detect deadcode (unreachable code)

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

**Comments about testing , should you have some** (optional)

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by smartprogrammer93 — APPROVED** [review]

File: src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs [hunk 1]
```diff
@@ -7,6 +7,9 @@
 using Nethermind.Evm;
 using Nethermind.Logging;
 using Nethermind.Specs;
+using static System.Collections.Specialized.BitVector32;
```
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs:10]: Remove these usings

File: src/Nethermind/Nethermind.Evm/GasCostOf.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public static class GasCostOf
         public const long RJumpv = 4;
         public const long Callf = 5;
         public const long Retf = 4;
-        public const long Jumpf = 4;
+        public const long Jumpf = 3;
```
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Evm/GasCostOf.cs:24]: yeet

File: src/Nethermind/Nethermind.Evm/Instruction.cs [hunk 3]
```diff
@@ -243,5 +281,13 @@ public static bool IsValid(this Instruction instruction, IReleaseSpec spec)
             Instruction.RETF or Instruction.CALLF => true,
             _ => false
         };
+
+        public static bool IsTerminatingInstruction(this Instruction instruction, IReleaseSpec spec = null) => instruction switch
```
Comment (smartprogrammer93) [inline src/Nethermind/Nethermind.Evm/Instruction.cs:285]: remove one of these. Redundant.

**Review by smartprogrammer93 — APPROVED** [review]

File: src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs [hunk 4]
```diff
@@ -466,5 +472,163 @@ bool ValidateInstructions(ReadOnlySpan<byte> code, in EofHeader header)
             }
             return true;
         }
+        public bool ValidateReachableCode(int sectionId, in ReadOnlySpan<byte> code, Dictionary<int, int>.KeyCollection reachedOpcode)
```
Comment (DemuirGos) [inline src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs:475]: Needs further optimazation

**Pull request closed without merging**